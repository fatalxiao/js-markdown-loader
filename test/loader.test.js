'use strict';

import {compile, getCompiler, getModuleSource, execute, readAsset, getWarnings, getErrors} from './helpers';

describe('loader', () => {

    it('images', async () => {

        const compiler = getCompiler('images.js');
        const stats = await compile(compiler);

        console.log('execute(readAsset(\'main.bundle.js\', compiler, stats))::', execute(readAsset('main.bundle.js', compiler, stats)));

        expect(getModuleSource('./images.js', stats)).toMatchSnapshot('module');
        expect(execute(readAsset('main.bundle.js', compiler, stats))).toMatchSnapshot('result');
        expect(getWarnings(stats)).toMatchSnapshot('warnings');
        expect(getErrors(stats)).toMatchSnapshot('errors');

    });

});
