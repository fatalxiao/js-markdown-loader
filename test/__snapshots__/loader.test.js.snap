// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader images: errors 1`] = `Array []`;

exports[`loader images: module 1`] = `
"import md from './images.md';

export default md;
"
`;

exports[`loader images: result 1`] = `
Object {
  "html": "<h1><a id=\\"Images\\" href=\\"#Images\\"></a>Images</h1><p><img alt=\\"Click Image\\" src=\\"/webpack/public/path/img/alcedo-logo.04a8d36.png\\"/></p>",
  "raw": "# Images

![Click Image](./alcedo-logo.png)
",
  "renderTree": Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "rawValue": "Images",
            "type": "Text",
          },
        ],
        "level": 1,
        "rawValue": "",
        "type": "AtxHeader",
      },
      Object {
        "rawValue": "",
        "type": "BlankLine",
      },
      Object {
        "children": Array [
          Object {
            "alt": "Click Image",
            "src": "./alcedo-logo.png",
            "type": "Image",
          },
        ],
        "rawValue": "",
        "type": "Paragraph",
      },
      Object {
        "rawValue": "",
        "type": "BlankLine",
      },
    ],
    "footnotes": Array [],
    "headerTree": Object {
      "children": Array [
        Object {
          "children": Array [],
          "innerHtml": "Images",
          "level": 1,
        },
      ],
      "isRoot": true,
    },
    "isRoot": true,
    "metaData": Object {},
    "referenceDefine": Object {},
  },
}
`;

exports[`loader images: warnings 1`] = `Array []`;

exports[`loader messageBoxes: errors 1`] = `Array []`;

exports[`loader messageBoxes: module 1`] = `
"'use strict';

import md from './messageBoxes.md';
"
`;

exports[`loader messageBoxes: result 1`] = `Object {}`;

exports[`loader messageBoxes: warnings 1`] = `Array []`;
